--- ../../src/left_main.php	2004-05-17 18:03:40.000000000 +0200
+++ ../../src/left_main.php	2004-06-01 12:31:50.000000000 +0200
@@ -50,7 +50,7 @@
 	$tmp_status = create_unseen_string($real_box, $box_array, $imapConnection, $unseen_type );
 	if ($status !== false) {
 	    $status = $tmp_status;
-	} 
+	}
     }
     list($unseen_string, $unseen) = $status;
     $special_color = ($use_special_folder_color && isSpecialMailbox($real_box));
@@ -65,13 +65,24 @@
     if ($status !== false) {
 	$line .= '<a href="right_main.php?PG_SHOWALL=0&amp;sort=0&amp;startMessage=1&amp;mailbox='.
                 $mailboxURL.'" TARGET="right" STYLE="text-decoration:none">';
-    } 
+    }
     if ($special_color) {
         $line .= "<font color=\"$color[11]\">";
     }
     if ( $mailbox == 'INBOX' ) {
+    		$line .= '<img src="../images/inbox.png" border="0" height="16" width="16" style="vertical-align: bottom;" alt="'. _("INBOX") .'" title="'. _("INBOX") .'">';
         $line .= _("INBOX");
+    } elseif ( $real_box == $draft_folder){
+    		$line .= '<img src="../images/draft.png" border="0" height="16" width="16" style="vertical-align: middle;" alt="'. $mailbox .'" title="'. $mailbox .'">';
+        $line .= str_replace(array(' ','<','>'),array('&nbsp;','&lt;','&gt;'),$mailbox);
+    } elseif ( $real_box == $sent_folder){
+    		$line .= '<img src="../images/senti.png" border="0" height="16" width="16" style="vertical-align: middle;" alt="'. $mailbox .'" title="'. $mailbox .'">';
+        $line .= str_replace(array(' ','<','>'),array('&nbsp;','&lt;','&gt;'),$mailbox);
+    } elseif ( $real_box == $trash_folder){
+    		$line .= '<img src="../images/delitem.png" border="0" height="16" width="16" style="vertical-align: middle;" alt="'. $mailbox .'" title="'. $mailbox .'">';
+        $line .= str_replace(array(' ','<','>'),array('&nbsp;','&lt;','&gt;'),$mailbox);
     } else {
+    		$line .= '<img src="../images/folder.png" border="0" height="16" width="16" style="vertical-align: middle;" alt="'. $mailbox .'" title="'. $mailbox .'">';
         $line .= str_replace(array(' ','<','>'),array('&nbsp;','&lt;','&gt;'),$mailbox);
     }
     if ($special_color == TRUE)
@@ -168,14 +179,14 @@
 function create_collapse_link($boxnum) {
     global $boxes, $imapConnection, $unseen_notify, $color;
     $mailbox = urlencode($boxes[$boxnum]['unformatted']);
-        
+
     /* Create the link for this collapse link. */
     $link = '<a target="left" style="text-decoration:none" ' .
             'href="left_main.php?';
     if ($boxes[$boxnum]['collapse'] == SM_BOX_COLLAPSED) {
-        $link .= "unfold=$mailbox\">+";
+        $link .= "unfold=$mailbox\"><IMG src=\"../images/plus.png\" border=\"0\" height=\"7\" width=\"7\">";
     } else {
-        $link .= "fold=$mailbox\">-";
+        $link .= "fold=$mailbox\"><IMG src=\"../images/minus.png\" border=\"0\" height=\"7\" width=\"7\">";
     }
     $link .= '</a>';
 
@@ -284,7 +295,7 @@
 function listBoxes ($boxes, $j=0 ) {
     global $data_dir, $username, $startmessage, $color, $unseen_notify, $unseen_type,
         $move_to_trash, $trash_folder, $collapse_folders, $use_special_folder_color;
-	
+
     $pre = '';
     $end = '';
     $collapse = false;
@@ -304,7 +315,7 @@
     	    if ($boxes->total) {
 		$numMessages = $boxes->total;
         	$unseen_string = "<font color=\"$color[11]\">($unseen/$numMessages)</font>";
-	    }	    
+	    }
 	} else $unseen = 0;
 
 
@@ -360,10 +371,10 @@
 	$fontend = '';
 	if ($use_special_folder_color && $boxes->is_special) {
     	    $font = "<font color=\"$color[11]\">";
-	    $fontend = "</font>";    
+	    $fontend = "</font>";
 	}
-	
-	if (!$boxes->is_root) { 
+
+	if (!$boxes->is_root) {
 	    echo "" . $pre .$font. str_replace(array(' ','<','>'),array('&nbsp;','&lt;','&gt;'),$boxes->mailboxname_sub) .$fontend . $end. '<br />';
 	    $j++;
 	}
@@ -371,8 +382,8 @@
     	    for ($i = 0; $i <count($boxes->mbxs); $i++) {
     		listBoxes($boxes->mbxs[$i],$j);
     	    }
-	}    
-	
+	}
+
     }
 }
 
@@ -386,7 +397,7 @@
     $pre = '';
     $end = '';
     $collapse = false;
-    
+
     if ($boxes) {
 	$mailbox = $boxes->mailboxname_full;
 	$mailboxURL = urlencode($mailbox);
@@ -399,7 +410,7 @@
     	    if ($boxes->total) {
 		$numMessages = $boxes->total;
         	$unseen_string = "<font color=\"$color[11]\">($unseen/$numMessages)</font>";
-	    }	    
+	    }
 	} else $unseen = 0;
 
 	/* If there are unseen message, bold the line. */
@@ -408,7 +419,7 @@
 	/* color special boxes */
 	if ($use_special_folder_color && $boxes->is_special) {
     	    $pre .= "<font color=\"$color[11]\">";
-	    $end .= '</font>';    
+	    $end .= '</font>';
 	}
 
 	/* If there are unseen message, close bolding. */
@@ -537,21 +548,21 @@
 }
 
 /**
- * $advanced_tree and $oldway are boolean vars which are default set to default 
- * SM behaviour. 
- * Setting $oldway to false causes left_main.php to use the new  experimental 
+ * $advanced_tree and $oldway are boolean vars which are default set to default
+ * SM behaviour.
+ * Setting $oldway to false causes left_main.php to use the new  experimental
  * way of getting the mailbox-tree.
- * Setting $advanced tree to true causes SM to display a experimental 
+ * Setting $advanced tree to true causes SM to display a experimental
  * mailbox-tree with dhtml behaviour.
- * It only works on browsers which supports css and javascript. The used 
- * javascript is experimental and doesn't support all browsers. It is tested on 
+ * It only works on browsers which supports css and javascript. The used
+ * javascript is experimental and doesn't support all browsers. It is tested on
  * IE6 an Konquerer 3.0.0-2.
  * In the function ListAdvancedBoxes there is another var $use_folder_images.
  * setting this to true is only usefull if the images exists in ../images.
- * 
+ *
  * Feel free to experiment with the code and report bugs and enhancements
  * to marc@its-projects.nl
- **/ 
+ **/
 
 $advanced_tree = false; /* set this to true if you want to see a nicer mailboxtree */
 $oldway = true; /* default SM behaviour */
@@ -594,10 +605,10 @@
 	          el.src="../images/plus.png";
                   document.getElementById(form_id).value=1;
 	       }
-	    }   
+	    }
 	}
    }
-   
+
    function preload() {
      if (!document.images) return;
      var ar = new Array();
@@ -606,8 +617,8 @@
         ar[i] = new Image();
 	ar[i].src = arguments[i];
      }
-   }        
-	    	  
+   }
+
    function buttonover(el,on) {
       if (!on) {
          el.style.borderColor="blue";}
@@ -616,14 +627,14 @@
    }
 
    function buttonclick(el,on) {
-      if (!on) { 
+      if (!on) {
          el.style.border="groove"}
       else {
          el.style.border="ridge";}
    }
 
    function hideframe(hide) {
-   
+
 ECHO;
 $xtra .= "      left_size = \"$left_size\";\n";
 $xtra .= <<<ECHO
@@ -639,7 +650,7 @@
 	leftbutton = document.getElementById("showf");
       } else {
         return false;
-      }	
+      }
       if(hide) {
          new_col = calc_col("20");
          masterf.cols = new_col;
@@ -652,16 +663,16 @@
 	 document.body.style.overflow='';
 	 leftbutton.style.display='none';
 	 leftcontent.style.display='block';
-	 
+
       }
    }
-   
+
    function calc_col(c_w) {
 
 ECHO;
    if ($location_of_bar == 'right') {
        $xtra .= '     right=true;';
-   } else {    
+   } else {
        $xtra .= '     right=false;';
    }
    $xtra .= "\n";
@@ -672,8 +683,8 @@
          new_col = c_w+',*';
      }
      return new_col;
-   }	 
-         
+   }
+
    function resizeframe(direction) {
      if (document.all) {
     	masterf = window.parent.document.all["fs1"];
@@ -682,32 +693,32 @@
      } else {
         return false;
      }
-     
+
 ECHO;
    if ($location_of_bar == 'right') {
        $xtra .= '  colPat=/^\*,(\d+)$/;';
-   } else {    
+   } else {
        $xtra .= '  colPat=/^(\d+),.*$/;';
    }
    $xtra .= "\n";
-  
+
 $xtra .= <<<ECHO
      old_col = masterf.cols;
      colPat.exec(old_col);
-     
+
      if (direction) {
         new_col_width = parseInt(RegExp.$1) + 25;
-	
+
      } else {
         if (parseInt(RegExp.$1) > 35) {
            new_col_width = parseInt(RegExp.$1) - 25;
         }
      }
-     masterf.cols = calc_col(new_col_width);	
+     masterf.cols = calc_col(new_col_width);
    }
 
 //-->
-   
+
 </script>
 
 ECHO;
@@ -832,7 +843,7 @@
 
 do_hook('left_main_before');
 if ($advanced_tree) {
-   /* nice future feature, needs layout !! volunteers?   */  
+   /* nice future feature, needs layout !! volunteers?   */
    $right_pos = $left_size - 20;
    echo '<div style="position:absolute;top:0;border=solid;border-width:0.1em;border-color:blue;"><div ID="hidef" style="width=20;font-size:12"><A HREF="javascript:hideframe(true)"><b><<</b></a></div>';
    echo '<div ID="showf" style="width=20;font-size:12;display:none;"><a href="javascript:hideframe(false)"><b>>></b></a></div>';
@@ -841,9 +852,9 @@
    echo '<div ID="leftframe"><br /><br />';
 }
 
-echo "\n\n" . html_tag( 'table', '', 'left', '', 'border="0" cellspacing="0" cellpadding="0" width="99%"' ) . 
-    html_tag( 'tr' ) . 
-    html_tag( 'td', '', 'left' ) . 
+echo "\n\n" . html_tag( 'table', '', 'left', '', 'border="0" cellspacing="0" cellpadding="0" width="99%"' ) .
+    html_tag( 'tr' ) .
+    html_tag( 'td', '', 'left' ) .
     '<center><font size="4"><b>'. _("Folders") . "</b><br /></font>\n\n";
 
 if ($date_format != 6) {
@@ -900,7 +911,7 @@
 }
 
 if ($oldway) {  /* normal behaviour SM */
- 
+
 sqgetGlobalVar('force_refresh',$force_refresh,SQ_GET);
 if (!isset($boxes)) { // auto_create_done
     $boxes = sqimap_mailbox_list($imapConnection,$force_refresh);
@@ -962,14 +973,14 @@
 
     }
 }
-} else {  /* expiremental code */ 
+} else {  /* expiremental code */
     $boxes = sqimap_mailbox_tree($imapConnection);
     if (isset($advanced_tree) && $advanced_tree) {
 	echo '<form name="collapse" action="left_main.php" method="post" ' .
             'enctype="multipart/form-data"'."\n";
 	echo '<small><button type="submit" class="button" onmouseover="buttonover(this,true)" onmouseout="buttonover(this,false)" onmousedown="buttonclick(this,true)" onmouseup="buttonclick(this,false)">'. _("Save folder tree") .'</button><br /><br />';
 	echo '<div id="mailboxes" class="mailboxes">'."\n\n";
-	if (!isset($mbx)) $mbx=NULL; 
+	if (!isset($mbx)) $mbx=NULL;
 	    ListAdvancedBoxes($boxes, $mbx);
 	echo '</div></small>'."\n";
 	echo '</form>'."\n";
--- ../../functions/mailbox_display.php 2004-04-19 18:21:53.000000000 +0200
+++ ../../functions/mailbox_display.php 2004-06-01 11:09:14.000000000 +0200
@@ -41,7 +41,7 @@
 function printMessageInfo($imapConnection, $t, $not_last=true, $key, $mailbox,
                           $start_msg, $where, $what) {
     global $checkall,
-           $color, $msgs, $msort, $td_str, $msg, 
+           $color, $msgs, $msort, $td_str, $msg,
            $default_use_priority,
            $message_highlight_list,
            $index_order,
@@ -78,7 +78,7 @@
        $msg['FROM'] = $msg['TO'];
     }
     $msg['FROM'] = parseAddress($msg['FROM'],1);
-    
+
        /*
         * This is done in case you're looking into Sent folders,
         * because you can have multiple receivers.
@@ -135,7 +135,7 @@
     } else {
         $searchstr = '';
     }
-    
+
     if (is_array($message_highlight_list) && count($message_highlight_list)) {
         $msg['TO'] = parseAddress($msg['TO']);
         $msg['CC'] = parseAddress($msg['CC']);
@@ -167,7 +167,7 @@
                             $headertest = strtolower(decodeHeader($msg[$match_type], true, false));
                             if (strstr('^^' . $headertest, $high_val)) {
                                 $hlt_color = $message_highlight_list_part['color'];
-                                break 3; 
+                                break 3;
                             }
                             break;
                     }
@@ -183,7 +183,7 @@
     $col = 0;
     $msg['SUBJECT'] = decodeHeader($msg['SUBJECT']);
     $subject = processSubject($msg['SUBJECT'], $indent_array[$msg['ID']]);
-    $subject = str_replace('&nbsp;',' ',$subject);    
+    $subject = str_replace('&nbsp;',' ',$subject);
     if (sizeof($index_order)) {
         foreach ($index_order as $index_order_part) {
             switch ($index_order_part) {
@@ -234,39 +234,60 @@
                 echo html_tag( 'td', $td_str, 'left', $hlt_color );
                 break;
             case 5: /* flags */
-                $stuff = false;
                 $td_str = "<b><small>";
 
-                if (isset($msg['FLAG_ANSWERED']) && $msg['FLAG_ANSWERED'] == true) {
-                    $td_str .= _("A");
-                    $stuff = true;
-                }
-                if ($msg['TYPE0'] == 'multipart') {
-                    $td_str .= '+';
-                    $stuff = true;
+                if (isset($msg['FLAG_FLAGGED']) && $msg['FLAG_FLAGGED'] == true) {
+                		$msg_alt = _("This message is Flagged");
+                    $td_str .= '<IMG SRC="../plugins/msg_flags/flagged.gif" border="0" height="10" width="10" alt="'. $msg_alt .'" title="'. $msg_alt .'"> ';
                 }
                 if ($default_use_priority) {
                     if ( ($msg['PRIORITY'] == 1) || ($msg['PRIORITY'] == 2) ) {
-                        $td_str .= "<font color=\"$color[1]\">!</font>";
-                        $stuff = true;
+                    		$msg_alt = _("This message is High Priority");
+                        $td_str .= '<IMG SRC="../plugins/msg_flags/prio_high.gif" border="0" height="10" width="5" alt="'. $msg_alt .'" title="'. $msg_alt .'"> ';
+                    }
+                    elseif ($msg['PRIORITY'] == 5) {
+                    		$msg_alt = _("This message is Low Priority");
+                        $td_str .= '<IMG SRC="../plugins/msg_flags/prio_low.gif" border="0" height="10" width="5" alt="'. $msg_alt .'" title="'. $msg_alt .'"> ';
                     }
-                    if ($msg['PRIORITY'] == 5) {
-                        $td_str .= "<font color=\"$color[8]\">?</font>";
-                        $stuff = true;
+                    else {
+                    		$td_str .= '<IMG SRC="../plugins/msg_flags/transparent.gif" border="0" height="10" width="5"> ';
                     }
                 }
-                if (isset($msg['FLAG_DELETED']) && $msg['FLAG_DELETED'] == true) {
-                    $td_str .= "<font color=\"$color[1]\">D</font>";
-                    $stuff = true;
+                if ($msg['TYPE0'] == 'multipart') {
+                		$msg_alt = _("This message is Mulitpart");
+                    $td_str .= '<IMG SRC="../plugins/msg_flags/attach.gif" border="0" height="10" width="6" alt="'. $msg_alt .'" title="'. $msg_alt .'">';
                 }
-                if (!$stuff) {
-                    $td_str .= '&nbsp;';
+                else {
+                		$td_str .= '<IMG SRC="../plugins/msg_flags/transparent.gif" border="0" height="10" width="6">';
                 }
+
+                $msg_icon = '';
+
+	              if (!isset($msg['FLAG_SEEN']) || $msg['FLAG_SEEN'] == true) {
+	              		$msg_alt = _("This message is Read");
+	                  $msg_icon .= "../plugins/msg_flags/msg_read";
+	              }
+	              else {
+	              		$msg_alt = _("This message is New");
+	                  $msg_icon .= '../plugins/msg_flags/msg_new';
+	              }
+	              if (isset($msg['FLAG_DELETED']) && $msg['FLAG_DELETED'] == true) {
+	              		$msg_alt .= _(" & Deleted");
+	                  $msg_icon .= '_deleted';
+	              }
+	              if (isset($msg['FLAG_ANSWERED']) && $msg['FLAG_ANSWERED'] == true) {
+	              		$msg_alt .= _(" & Answered");
+	                  $msg_icon .= '_reply';
+	              }
+
+         				$td_str .= '<IMG SRC="' . $msg_icon . '.gif" border="0" alt="'. $msg_alt . '" title="' . $msg_alt . '" height="12" width="18" ></small></b>';
+
+                $td_str .= '&nbsp;';
                 do_hook("msg_envelope");
                 $td_str .= '</small></b>';
                 echo html_tag( 'td',
                                $td_str,
-                               'center',
+                               'right',
                                $hlt_color,
                                'nowrap' );
                 break;
@@ -299,7 +320,7 @@
             $end_loop = $num_msgs - $start_msg + 1;
         } else {
             $end_loop = $show_num;
-        }
+        }
         return fillMessageArray($imapConnection,$id,$end_loop,$show_num);
     } else {
         return false;
@@ -350,7 +371,7 @@
             } else {
                 $end_loop = $show_num;
             }
-        }
+        }
         $msgs = fillMessageArray($imapConnection,$id,$end_loop, $show_num);
     }
     return $msgs;
@@ -424,7 +445,7 @@
 	    sqgetGlobalVar('msort', $msort, SQ_SESSION);
 	} else {
     	    sqsession_unregister('msort');
-    	    sqsession_unregister('msgs');
+    	    sqsession_unregister('msgs');
 	}
         switch ($mode) {
             case 'thread':
@@ -483,11 +504,11 @@
     echo '<table border="0" width="100%" cellpadding="0" cellspacing="0">';
     echo '<tr><td>';
 
-    mail_message_listing_beginning($imapConnection, $mailbox, $sort, 
+    mail_message_listing_beginning($imapConnection, $mailbox, $sort,
                                   $msg_cnt_str, $paginator_str, $start_msg);
     echo '</td></tr>';
     /* line between the button area and the list */
-    echo '<tr><td HEIGHT="5" BGCOLOR="'.$color[4].'"></td></tr>';  
+    echo '<tr><td HEIGHT="5" BGCOLOR="'.$color[4].'"></td></tr>';
 
     echo '<tr><td>';
     echo '    <table width="100%" cellpadding="1" cellspacing="0" align="center"'.' border="0" bgcolor="'.$color[9].'">';
@@ -496,11 +517,11 @@
     echo '<tr><td>';
     printHeader($mailbox, $srt, $color, !$thread_sort_messages);
 
-    displayMessageArray($imapConnection, $num_msgs, $start_msg, 
+    displayMessageArray($imapConnection, $num_msgs, $start_msg,
 		      $msort, $mailbox, $sort, $color, $show_num,0,0);
     echo '</td></tr></table></td></tr></table>';
 
-    mail_message_listing_end($num_msgs, $paginator_str, $msg_cnt_str, $color); 
+    mail_message_listing_end($num_msgs, $paginator_str, $msg_cnt_str, $color);
     echo '</td></tr></table>';
     //$t = elapsed($start);
     //echo("elapsed time = $t seconds\n");
@@ -690,9 +711,9 @@
                     html_tag( 'table' ,
                         html_tag( 'tr',
                             html_tag( 'td', $paginator, 'left' ) .
-                            html_tag( 'td', $msg_cnt_str, 'right' ) 
+                            html_tag( 'td', $msg_cnt_str, 'right' )
                         )
-                    , '', $color[4], 'border="0" width="100%" cellpadding="1"  cellspacing="0"' ) 
+                    , '', $color[4], 'border="0" width="100%" cellpadding="1"  cellspacing="0"' )
                 , 'left', '', '' )
             , '', $color[0] )
 	    , '', '', 'border="0" width="100%" cellpadding="1"  cellspacing="0"' );
@@ -711,9 +732,9 @@
             )
             . html_tag( 'tr' ) ."\n"
             . html_tag( 'td', '', 'left', '', 'valign="middle" nowrap' );
-            getMbxList($imapConnection);  
-            echo getButton('SUBMIT', 'moveButton',_("Move")) . "\n";   
-            echo getButton('SUBMIT', 'attache',_("Forward")) . "\n";   
+            getMbxList($imapConnection);
+            echo getButton('SUBMIT', 'moveButton',_("Move")) . "\n";
+            echo getButton('SUBMIT', 'attache',_("Forward")) . "\n";
 
   echo "      </small></td>\n"
          . html_tag( 'td', '', 'right', '', 'nowrap' );
@@ -725,6 +746,8 @@
              .'&nbsp;' . _("mailbox") . "\n";
     }
 
+		echo getButton('SUBMIT', 'markFlagged',_("Flag"));
+    echo getButton('SUBMIT', 'markUnflagged',_("Unflag"));
     echo getButton('SUBMIT', 'markRead',_("Read"));
     echo getButton('SUBMIT', 'markUnread',_("Unread"));
     echo getButton('SUBMIT', 'delete',_("Delete")) ."&nbsp;\n";
@@ -776,7 +799,7 @@
 function mail_message_listing_end($num_msgs, $paginator_str, $msg_cnt_str, $color) {
   if ($num_msgs) {
     /* space between list and footer */
-    echo '<tr><td HEIGHT="5" BGCOLOR="'.$color[4].'" COLSPAN="1">';  
+    echo '<tr><td HEIGHT="5" BGCOLOR="'.$color[4].'" COLSPAN="1">';
 
     echo '</td></tr><tr><td>';
     echo html_tag( 'table',
@@ -808,7 +831,7 @@
     $widths = array(1=>1,2=>25,3=>5,4=>0,5=>1,6=>5);
     $subjectwidth = 100;
     foreach($index_order as $item) {
-        $subjectwidth -= $widths[$item]; 
+        $subjectwidth -= $widths[$item];
     }
 
     foreach ($index_order as $item) {
@@ -1103,7 +1126,7 @@
 	 */
          // echo "qts =  $q1_pgs/$q2_pgs/$q3_pgs/$q4_pgs = "
          //     . ($q1_pgs + $q2_pgs + $q3_pgs + $q4_pgs) . '<br>';
-         
+
 
         /* Print out the page links from the compute page quarters. */
 
@@ -1209,7 +1232,7 @@
      * see if this is entities-encoded string
      * If so, Iterate through the whole string, find out
      * the real number of characters, and if more
-     * than 55, substr with an updated trim value. 
+     * than 55, substr with an updated trim value.
      */
     $step = $ent_loc = 0;
     while ( $ent_loc < $trim_val && (($ent_loc = strpos($subject, '&', $ent_offset)) !== false) &&
@@ -1218,7 +1241,7 @@
         $ent_offset  = $ent_loc_end+1;
         ++$step;
     }
-    
+
     if (($trim_val > 50) && (strlen($subject) > ($trim_val))&& (strpos($subject,';',$trim_val) < ($trim_val +6))) {
         $i = strpos($subject,';',$trim_val);
         if ($i) {
@@ -1245,7 +1268,7 @@
 function getMbxList($imapConnection) {
     global $lastTargetMailbox;
     echo  '         <small>&nbsp;<tt><select name="targetMailbox">';
-    echo sqimap_mailbox_option_list($imapConnection, array(strtolower($lastTargetMailbox)) ); 
+    echo sqimap_mailbox_option_list($imapConnection, array(strtolower($lastTargetMailbox)) );
     echo '         </SELECT></TT>&nbsp;';
 }
 
@@ -1279,7 +1302,7 @@
 
 function handleAsSent($mailbox) {
     global $handleAsSent_result;
- 
+
     /* First check if this is the sent or draft folder. */
     $handleAsSent_result = isSentMailbox($mailbox) || isDraftMailbox($mailbox);
 
--- ../../src/move_messages.php	2004-05-03 23:39:35.000000000 +0200
+++ ../../src/move_messages.php	2004-06-01 09:54:03.000000000 +0200
@@ -129,6 +129,9 @@
 sqgetGlobalVar('markUnread',      $markUnread,      SQ_POST);
 sqgetGlobalVar('attache',         $attache,         SQ_POST);
 sqgetGlobalVar('location',        $location,        SQ_POST);
+sqgetGlobalVar('markFlagged',     $markFlagged,     SQ_POST); /* Added for Msg_Flags */
+sqgetGlobalVar('markUnflagged',   $markUnflagged,   SQ_POST); /* Added for Msg_Flags */
+
 
 /* end of get globals */
 
@@ -190,6 +193,10 @@
                 sqimap_toggle_flag($imapConnection, $id, '\\Seen',true,true);
             } else if (isset($markUnread)) {
                 sqimap_toggle_flag($imapConnection, $id, '\\Seen',false,true);
+            } else if (isset($markFlagged)) {
+              	sqimap_toggle_flag($imapConnection, $id, '\\Flagged',true,true);
+          	} else if (isset($markUnflagged)) {
+            		sqimap_toggle_flag($imapConnection, $id, '\\Flagged',false,true);
             } else  {
                 sqimap_msgs_list_delete($imapConnection, $mailbox, $id);
                 if ($auto_expunge) {
