Description: imscp patch
Author: i-MSCP Team <team@i-mscp.net>

--- vsftpd-3.0.2.orig/defs.h
+++ vsftpd-3.0.2/defs.h
@@ -19,7 +19,7 @@
 /* Must be at least the size of VSFTP_MAX_COMMAND_LINE, VSFTP_DIR_BUFSIZE and
    VSFTP_DATA_BUFSIZE*2 */
 #define VSFTP_PRIVSOCK_MAXSTR   VSFTP_DATA_BUFSIZE * 2
-#define VSFTP_AS_LIMIT          100UL * 1024 * 1024
+#define VSFTP_AS_LIMIT          400UL * 1024 * 1024

 #endif /* VSF_DEFS_H */
 
