--- plugins/archive_mail/includes/archive_mail_bottom.php.orig	Mon Jul 26 02:19:45 2004
+++ plugins/archive_mail/includes/archive_mail_bottom.php	Mon Jul 26 03:55:34 2004
@@ -8,9 +8,16 @@
     Version ........ 1.2
     Purpose ........ Allows you to download your email in a compressed archive
 
+    Hacked to co-exist with Preview Pane plugin under SM 1.4.x.
+
 *******************************************************************************/
 
-      global $mailbox, $imapConnection, $SQM_INTERNAL_VERSION;
+      global $mailbox, $imapConnection, $SQM_INTERNAL_VERSION, 
+             $archive_mail_button_has_been_printed, 
+             $archiveMailIsPatchedForPreviewPanePlugin;
+
+      $archiveMailIsPatchedForPreviewPanePlugin = 1;
+
       if (isset($SQM_INTERNAL_VERSION) && $SQM_INTERNAL_VERSION[1] == 5)
          return;
 
@@ -28,14 +35,17 @@
                echo _("Archive disabled (zlib not supported)");
                print "</td></tr></table>";
             } else {
-               print "\n<tr width=\"100%\"><td><p align=right><input type=submit NAME=\"archiveButton\" value='";
+               print "\n<tr width=\"100%\"><td align=\"right\"><input type=\"submit\" NAME=\"archiveButton\" value='";
                echo _("Archive");
-               print "'></p></td></tr>\n";
+               print "'>"; // </td></tr>\n";
             }
 
 	    // Unset domain
 	    bindtextdomain('squirrelmail', SM_PATH . 'locale');
 	    textdomain('squirrelmail');
+
+            $archive_mail_button_has_been_printed = 1;
+
          }
       }
-?>
\ No newline at end of file
+?>
