Index: library/Zend/Validate/File/Count.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- a/library/Zend/Validate/File/Count.php	(revision 136735e776f520b081cd374012852cb88cef9a88)
+++ b/library/Zend/Validate/File/Count.php	(date 1561371014000)
@@ -254,7 +254,12 @@
             $this->addFile($value);
         }
 
-        $this->_count = count($this->_files);
+        if (NULL !== $this->_files) {
+            $this->_count = count($this->_files);
+        } else {
+            $this->_count = 0;
+        }
+
         if (($this->_max !== null) && ($this->_count > $this->_max)) {
             return $this->_throw($file, self::TOO_MANY);
         }
